File: src/dotnet/runner/DotnetRunner/ITest.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace DotnetRunner
{
    interface ITest<TInput, TOutput>
    {
        // This function must be called before any other function.
        void Prepare(TInput input);
        // calculate function
        void CalculateObjective(int times);
        void CalculateJacobian(int times);
        TOutput Output();
    }
}

File: src/python/shared/ITest.py
# Copyright (c) Microsoft Corporation.
# Licensed under the MIT license.

from abc import ABC, abstractmethod

class ITest(ABC):
    # This function must be called before any other function.
    @abstractmethod
    def prepare(self, input):
        pass 

    @abstractmethod
    def calculate_objective(self, times):
        pass

    @abstractmethod
    def calculate_jacobian(self, times):
        pass

    @abstractmethod
    def output(self):
        pass